### DISCUSSION TITLE
Remember last track in playlist and resume again

### DISCUSSION THREAD

**Discussion by Tonitom** [hklages/node-red-contrib-sonos-plus#183] [Q&A] [answered]
Hello,
I've been waiting a long time for a long missed feature in music streaming services, I want to pick up a playlist where I left off listening. With Sonos-Plus this seems to be within my reach.

Does anyone possibly already have a flow that can do that?

Tonitom

**Comment by hklages** [comment]
Hello. 
I guess that is already now possible. If you play playlist, stopp it and play again, it continues at that position. 
Or do you have a collection of playlist and want to remember for each the last state? 

Many things are possible with sonos-plus and with sonos-event. For instance, you could "watch and store" the current playlist and position provided by the sonos-event and store it in a flow /global variable for later access. 

Tell me your use case and we can work on it. I will return from vacation mid of December.

**Comment by Tonitom** [comment]
Hi, 
in Spotify, there are playlists of radio plays filled with several hundred tracks. For these playlists, it would be great to be able to save the old playback position so that you can continue listening later, even if you have listened to something else in the meantime.

I've seen that you can read the current queue, but haven't found a way to figure out how to find the current position.

**Comment by hklages** [comment, marked as answer]
Hi, please have a look at group.get.trackplus. There you should find all information. group.play.track plays a specific track in queue.

**Comment by Tonitom** [comment]
I have looked at both, looks promising, tonight I will build that.

I would now trigger this every few seconds, or is there a more elegant variant? The events do not update as I imagine. Which event reacts when the track changes?

And thank you for your support, you safe me allot of time.

**Comment by hklages** [comment]
This node works well together with nodered-contrib-sonos-events. With the
event node you can listen to changes in the content

**Comment by Tonitom** [comment]
Nice, it works fine. 

When I tested sonos-events, the listener only monitored the leader of the group. Now I monitor all of them and everything works as I hoped.

When an event is detected, the playlist title is checked, if it matches, then the postion number is stored in a global variable. The next time I play the playlist, it continues at the last track.

**Comment by hklages** [comment]
Glad to hear and thanks for the feedback.

**Reply by Tonitom** [reply]
Hello again,
is there a way to extend the timeout for group.play.mysonos? The 3000 track playlist times out and cannot trigger group.play.track. I have a workaround with a fixed delay, but it would be nicer otherwise.

Tonitom

**Comment by hklages** [comment]
See #178 
I am going to work on it late of December.

**Comment by Tonitom** [comment]
Works amazingly well.

I now have a homemade NFC reader in every Sonos room, the kids have several NFC tags of their favourite stories and I have the tags of my favourite playlists. 
I can put the tag on any reader and the playback starts with the last track I listened to in the room where I used the reader. 

I've wanted to do something like that for years. A really big thank you to your project.

Tonitom

**Comment by hklages** [comment]
You are welcome. Really nice project!
